operantanalysis/operantanalysis.py

```diff
@@ -7,15 +8,14 @@ def accessfiles(filename):
     :param filename: string that refers to operant file location, file is txt
     :return: subject number, list of time codes, and list of eventcodes
     """
-    fileref = open(filename, "r")
-    filestring = fileref.read()
-    fileref.close()
+    with open(filename, "r") as fileref:
+        filestring = fileref.read()
 
     subjectlocation = filestring.index('Subject:')
-    subjectnumber = filestring[subjectlocation+9:subjectlocation+15]
+    subjectnumber = filestring[subjectlocation + 9:subjectlocation + 15]
 
     warraybegin = filestring.index('W:')
-    timeeventcodes = filestring[warraybegin+1:].split()[2:]
+    timeeventcodes = filestring[warraybegin + 1:].split()[2:]
```

Comment:
Having looked at a sample file you have -- I would highly recommend to refactor this code to make it more generic for reading files of that format.  What you really want is to implement 
- read lines from a file
- if next line starts not with space - it means you got a new field
- each field entry is `<Name>: <Value>`  .  `<Value>` could be on the same line or contained in the following lines until next field (so until next line which doesn't start with space) or end of file

as a result of such function you can return an ordereddict with all those fields

then you can define the "schema" for data types expected for each field and parse fields correspondingly -- like A and W have those vectors, etc

then you could have tiny unittests where you fabricate your interesting/tricky tiny input files and guarantee that what you read is what you expect to be read.

Reply:
Added this idea to most recent branch